src/lib/form/field.tsx

```diff
@@ -80,14 +84,67 @@ const StyledInput = styled.input<{
   padding-top: 14px;
   padding-bottom: 14px;
   padding-left: 16px;
-  padding-right: ${({ Icon, variant }) => {
-    if (Icon) return "56px";
-    if (variant) return "44px";
-    return "16px";
+  padding-right: ${({ Icon, variant, type }) => {
+    if (Icon) return type === "number" ? "64px" : "56px";
+    if (variant) return type === "number" ? "52px" : "44px";
+    return type === "number" ? "30px" : "16px";
   }};
+
+  /* Chrome, Safari, Edge, Opera */
+  ::-webkit-outer-spin-button,
+  ::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  /* Firefox */
+  -moz-appearance: textfield;
+  appearance: textfield;
   ${baseInputStyle}
 `;
 
+const ArrowsContainer = styled.div<{
+  variant?: string;
+  Icon?: React.FC<React.SVGAttributes<SVGElement>>;
+}>`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  position: absolute;
+  height: 17px;
+  width: 14px;
+  top: 14px;
```

Comment:
I would add a 4px gap to improve usability on mobile, without the gap I think the buttons are too close and the user can easily missclick. I would also improve the vertical alignment making the container 100% of the height and then adding the `justify-content: center`
```suggestion
  flex-direction: column;
  gap: 4px;
  align-items: center;
  justify-content: center;
  position: absolute;
  height: 100%;
  width: 14px;
  top: 0px;
```